src/signed.rs

```diff
@@ -17,10 +17,11 @@ impl<T: Debug + Clone + Encode + Decode + Send + 'static> Signature for T {}
 /// The meaning of sign is then to produce a signature `s` using the given private key,
 /// and `verify(msg, s, j)` is to verify whether the signature s under the message msg is
 /// correct with respect to the public key of the jth node.
+#[async_trait]
 pub trait KeyBox: Index + Clone + Send + Sync + 'static {
     type Signature: Signature;
     /// Signs a message `msg`.
-    fn sign(&self, msg: &[u8]) -> Self::Signature;
+    async fn sign(&self, msg: &[u8]) -> Self::Signature;
     /// Verifies whether a node with `index` correctly signed the message `msg`.
     fn verify(&self, msg: &[u8], sgn: &Self::Signature, index: NodeIndex) -> bool;
```

Comment:
If `sign` is already `async` shouldn't `verify` also be? The access the same resources in very similar ways, so it's quite a bit strange that one is and one isn't.

Reply:
In substrate, public keys are stored in-memory while the private keys can be stored in the filesystem (https://crates.parity.io/sc_keystore/struct.LocalKeystore.html), this is why only signing is `async`.